### PR TITLE
Node v18 upgrade

### PR DESCRIPTION
**Pull request by ethitter**
### Make sure you complete these items:

- [x] Updated root CHANGELOG.md with summary of changes under `Unpublished` section
- [x] `npm run prod` in this repo outputs expected changes (excepting the issue with re-ordered partials in larva-css algorithms partials - see [LRVA-1885](https://jira.pmcdev.io/browse/LRVA-1885))
- [ ] If adding a new pattern, in the PR comment, included a screenshot and link to the static Vercel deployment
- [ ] If changes to build scripts or the Node.js server, tested changes in pmc-spark [via a pre-release](https://confluence.pmcdev.io/x/XhOeAw)
- - If changes to build tools: npm scripts `prod`, `lint`, and `dev` scripts run as expected
- - If changes to Larva server: Static site generates as expected in a theme  (avail. on a URL {brand}.stg.larva.pmcdev.io)

### CODE REVIEW

**Comment by eboniebutler** [comment]
@ethitter Do we want to compile the files after the version bumps?

**Comment by ethitter** [comment]
@eboniebutler That's an excellent question; I'm torn. This PR is already going to be gigantic, but we also need to know that all of the packages still work.

A compromise is to test each package's build locally and fix any incompatibilities in the build scripts themselves, but wait to commit new built files until after this PR is merged. Once merged, we can create another PR that encompasses all of the rebuilds. That approach will make the version history a bit less chaotic, and allows us to roll back the builds independently of the Node upgrades should we discover some catastrophic regression.

Thoughts?

File: packages/larva-css/gulpfile.js [hunk 1]
```diff
@@ -8,10 +8,9 @@
  * dest file according to the changed file.
  */
 
-const Fiber = require( 'fibers' );
```
Comment (JordanPak) [inline packages/larva-css/gulpfile.js:11]: `fibers` is no longer required.

File: packages/larva-css/gulpfile.js [hunk 2]
```diff
@@ -8,10 +8,9 @@
...
 const gulp = require( 'gulp' );
 const path = require( 'path' );
-const sass = require( 'gulp-sass' );
+const sass = require( 'gulp-sass' )( require( 'sass' ) );
```
Comment (JordanPak) [inline packages/larva-css/gulpfile.js:13]: `gulp-sass` wants you to "bring your own SASS" in newer versions.

File: packages/larva/bin/larva.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ const hasScriptFile = fs.existsSync(
 );
 
 if ( hasScriptFile ) {
+	console.log( `Running ${ chalk.italic( scriptName ) } script\n` );
```
Comment (JordanPak) [inline packages/larva/bin/larva.js:18]: Adds clarity to what's going on

File: packages/larva/lib/utils/getPatternData.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
-const chalk = require( 'chalk' );
```
Comment (JordanPak) [inline packages/larva/lib/utils/getPatternData.js:1]: Wasn't being used

File: packages/larva/lib/utils/getSubDirectoryNames.js [hunk 5]
```diff
@@ -1,5 +1,4 @@
 const fs = require( 'fs' );
-const chalk = require( 'chalk' );
```
Comment (JordanPak) [inline packages/larva/lib/utils/getSubDirectoryNames.js:2]: Wasn't being used

File: packages/site/src/components/App.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
+import { BrowserRouter as Router, Routes, Route, Link } from 'react-router-dom';
```
Comment (JordanPak) [inline packages/site/src/components/App.js:2]: `Switch` was deprecated in favor of `Routes`.

File: packages/site/src/components/TokensView.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import { Route, Switch, useRouteMatch } from 'react-router-dom';
+import { Route, Routes, useMatch } from 'react-router-dom';
```
Comment (JordanPak) [inline packages/site/src/components/TokensView.js:1]: `useRouteMatch` was deprecated in favor of `useMatch()` in the newer version of React.

**Pull request merged**